### PR TITLE
Misc. readability improvements to createStackAtlas

### PR DESCRIPTION
**Pull request by 0xdaryl**
* reorder and group initialization code more logically
* rename variables to improve readability
* add additional documentation
* misc code formatting changes
* remove unneeded include directives

None of these changes the functionality or the order of operations
in this function.

Signed-off-by: Daryl Maier <maier@ca.ibm.com>

### CODE REVIEW

**Comment by 0xdaryl** [comment]
Jenkins test sanity all jdk8,jdk11

**Comment by 0xdaryl** [comment]
Jenkins test sanity.functional all jdk11

**Comment by 0xdaryl** [comment]
Jenkins test sanity.functional win,osx jdk8

**Comment by 0xdaryl** [comment]
Jenkins test sanity all jdk11

**Comment by 0xdaryl** [comment]
Jenkins test sanity all jdk8

**Comment by 0xdaryl** [comment]
The Linux failure is #4657.  The Windows failure is the infra issue #6167.

**Comment by 0xdaryl** [comment]
I'd appreciate a review/merge from one of @ymanton @fjeremic @gita-omr @dsouzai  please.

File: runtime/compiler/codegen/CodeGenGC.cpp [hunk 1]
```diff
@@ -71,117 +68,125 @@ J9::CodeGenerator::createStackAtlas()
    //
    TR::Compilation *comp = self()->comp();
    TR::ResolvedMethodSymbol * methodSymbol = comp->getMethodSymbol();
-   int32_t slotIndex = 0;
 
-   intptrj_t stackSlotSize = TR::Compiler->om.sizeofReferenceAddress();
+   bool doLocalsCompaction = (self()->getLocalsIG() && self()->getSupportsCompactedLocals()) ? true : false;
```
Comment (fjeremic) [inline runtime/compiler/codegen/CodeGenGC.cpp]: I'm sure you're doing a ton more work in this area, so perhaps we can fix this one on the next round. How about:

```
const bool doLocalsCompaction = self()->getLocalsIG() && self()->getSupportsCompactedLocals();
```

File: runtime/compiler/codegen/CodeGenGC.cpp [hunk 2]
```diff
@@ -71,117 +68,125 @@ J9::CodeGenerator::createStackAtlas()
...
    //
-   TR_GCStackMap * parameterMap = new (self()->trHeapMemory(), numberOfParmSlots) TR_GCStackMap(numberOfParmSlots);
+   int32_t slotIndex = 0;
+
    parameterIterator.reset();
    for (parmCursor = parameterIterator.getFirst(); parmCursor; parmCursor = parameterIterator.getNext())
       {
-      // always has to be on stack for FSD
-      if (interpreterFrameShape || comp->getOption(TR_FullSpeedDebug))
+      if (mimicInterpreterFrameShape || comp->getOption(TR_FullSpeedDebug))
```
Comment (fjeremic) [inline runtime/compiler/codegen/CodeGenGC.cpp]: This is a little inconsistent. We privatized `comp->getOption(TR_MimicInterpreterFrameShape)` but did not privatize `comp->getOption(TR_FullSpeedDebug)`. I would prefer we just use the actual `getOption` everywhere as it makes grepping for code guarded by options much easier.

File: runtime/compiler/codegen/CodeGenGC.cpp [hunk 3]
```diff
@@ -266,6 +268,24 @@ J9::CodeGenerator::createStackAtlas()
          }
       }
 
+   // Iniialize colour mapping
+   //
+   int32_t *colourToGCIndexMap = 0;
```
Comment (fjeremic) [inline runtime/compiler/codegen/CodeGenGC.cpp]: Is this related to colouring RA? I think we deprecated that. I'd like to understand more about this.

Comment (0xdaryl) [reply runtime/compiler/codegen/CodeGenGC.cpp]: No, this is for the CompactLocals optimization.  It uses graph colouring to reduce the stack space requirements by sharing storage for non-interfering locals.

**Comment by 0xdaryl** [comment]
I agree with your points.  Given the hand-holding I had to do to get the tests to pass (infra and known intermittent issues) up to this point, I'd like to address them in a separate PR and merge this as-is.
